plugins/mac-avcapture/av-capture.mm

```diff
@@ -2236,7 +2237,28 @@ static void add_manual_properties(obs_properties_t *props)
 		OBS_COMBO_FORMAT_STRING);
 	obs_property_list_add_string(dev_list, "", "");
 
-	for (AVCaptureDevice *dev in [AVCaptureDevice devices]) {
+	NSArray *devices = nil;
+#if (__MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_15)
+	if (@available(macOS 10.15, *)) {
+		AVCaptureDeviceDiscoverySession *deviceDiscoverySession = [AVCaptureDeviceDiscoverySession
+			discoverySessionWithDeviceTypes:@[
+				AVCaptureDeviceTypeBuiltInWideAngleCamera,
+				AVCaptureDeviceTypeExternalUnknown
+			]
```

Comment:
Could we include DeskViewCamera here?
Documentation implies we might not (`You can use this device type with AVCaptureMultiCamSession`), but in other cases of special cameras the docs explicitly say `You can only use this...`, so I'm unsure (and I don't have an iOS device to test).

Reply:
According to the docs, only valid `AVCaptureDeviceType` values are allowed, can be one of:

* AVCaptureDeviceTypeBuiltInDualCamera
* AVCaptureDeviceTypeBuiltInDualWideCamera
* AVCaptureDeviceTypeBuiltInTripleCamera
* AVCaptureDeviceTypeBuiltInWideAngleCamera
* AVCaptureDeviceTypeBuiltInUltraWideCamera
* AVCaptureDeviceTypeBuiltInTelephotoCamera
* AVCaptureDeviceTypeBuiltInLiDARDepthCamera
* AVCaptureDeviceTypeBuiltInTrueDepthCamera
* AVCaptureDeviceTypeBuiltInMicrophone
* AVCaptureDeviceTypeExternalUnknown

(See also: https://developer.apple.com/documentation/avfoundation/avcapturedevicetype)

Reply:
Discussed off-thread that this can be implemented later.